velox/type/DecimalUtil.h

```diff
@@ -54,8 +45,8 @@ class DecimalUtil {
     int128_t rescaledValue = inputValue.unscaledValue();
     auto scaleDifference = toScale - fromScale;
     if (scaleDifference >= 0) {
-      rescaledValue =
-          mul(rescaledValue, DecimalUtil::kPowersOfTen[scaleDifference]);
+      rescaledValue = checkedMultiply<int128_t>(
```

Comment:
this should be `mul`

Reply:
I will switch this *, since sanitize bug is fixed.

Reply:
> I will switch this *, since sanitize bug is fixed.

Ignore this. We need checkedMultiply here as well for corner cases. Added a test in CastExprTest as well.

Reply:
can we move this to another PR?